looting blacklist

divinity76 opened the pull request:
When "loot every item" is off, it will loot everything in the loot list.

When "loot every item" is on, it will loot everything that is not in the loot list, effectively making the loot list a blacklist when "loot every item" is on.

Related upstream issue: https://github.com/Vithrax/vBot/issues/15

I have tried submitting this upstream too, but it seems vBot upstream is unmaintained: https://github.com/Vithrax/vBot/pull/18

divinity76 committed:
looting blacklist

When "loot every item" is off, it will loot everything in the loot list.

When "loot every item" is on, it will loot everything that is not in the loot list, effectively making the loot list a blacklist when "loot every item" is on.

divinity76 committed:
blacklist UI

Commit:
Merge branch 'main' into patch-2

kokekanon commented:
@divinity76 

# 1) Should you change this text as well? 
![image](https://github.com/user-attachments/assets/c1f7ea27-20db-4a45-9731-e1163c095833)


`@onClick`

Example
![fix drop](https://github.com/user-attachments/assets/2d881485-cee4-4700-b0f1-6ac8e98c550c)


**it is just an example, I think it is inverted**.

.\bot\vBot_4.8\targetbot\looting.lua
```
  ui.everyItem.onClick = function()
    ui.everyItem:setOn(not ui.everyItem:isOn())
    if ui.everyItem:isOn() then
      ui.labelToLoot:setText("Items to loot")
    else
      ui.labelToLoot:setText("Items to ignore")
    end
    TargetBot.save()
  end
```
.bot\vBot_4.8\targetbot\looting.otui
```
  Label
    id: labelToLoot
    margin-top: 5
    text: Items to loot
    text-align: center
```
you could also change the text on the button 

--------------------------------------------

# 2) I suggest changing the font, because the text exceeds the button.

![image](https://github.com/user-attachments/assets/d32949fa-d75b-472d-8595-448ff02e4beb)

to
```
  BotSwitch
    id: everyItem
    !text: tr("Loot every item, except these")
    font: small-9px <----
    margin-top: 2

```

![image](https://github.com/user-attachments/assets/c1f6c556-70e5-4c97-be41-c7c990f5d66f)

Commit:
Merge branch 'main' into patch-2

divinity76 committed:
UI improvements from PR review

suggested by @kokekanon at https://github.com/mehah/otclient/pull/881#issuecomment-2386620565

divinity76 committed:
oops inverted it

divinity76 committed:
*mumble*

divinity76 commented:
@kokekanon that's better! thanks, added :)

Commit:
Merge branch 'main' into patch-2
